app/services/block_domain_service.rb

```diff
@@ -47,11 +47,15 @@ def suspend_accounts!
   def notify_of_severed_relationships!
     return if @domain_block_event.nil?
 
-    # TODO: check how efficient that query is, also check `push_bulk`/`perform_bulk`
+    notification_jobs_args = []
+
     @domain_block_event.affected_local_accounts.reorder(nil).find_each do |account|
       event = AccountRelationshipSeveranceEvent.create!(account: account, relationship_severance_event: @domain_block_event)
-      LocalNotificationWorker.perform_async(account.id, event.id, 'AccountRelationshipSeveranceEvent', 'severed_relationships')
+      notification_jobs_args.push([account.id, event.id, 'AccountRelationshipSeveranceEvent', 'severed_relationships'])
     end
+
+    # perform_bulk defaults to 1000 size batches, same as find_each
+    LocalNotificationWorker.perform_bulk(notification_jobs_args)
```

Comment:
Thank you for your contribution!

This could grow into a pretty large in-memory `notification_jobs_args` array. If going this way, I'd probably use `find_in_batches` and build an array for each such group instead of for all affected groups. As you pointed out, `find_each` works with the same batch size as `perform_bulk`, so this would not actually change how jobs are sent to Sidekiq, it would just put a bound on the size of the in-memory array.

Reply:
That's a great point. We can and should indeed use `find_in_batches` if going this way. I've just pushed an adjusted implementation that should retain the benefits of 1000 batch Redis/job queue pushes while also limiting the in-memory array sizes via the aforementioned batch method.

Reply:
Looks good, thanks!